src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,117 @@
+package frc.team2412.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+
+public class IntakeSubsystem extends SubsystemBase {
+
+    // Constants
+
+    public static class IntakeConstants {
+        public static String teamColor = "placeholder";
+        public static final double INTAKE_SPEED = 1;
+
+    }
+
+    // Enums
+
+    public enum IntakeMotorState {
+        IN, OUT, STOPPED;
+    }
+
+    public enum IntakeSolenoidState{
+        EXTENDED, RETRACTED;
+    }
+    
+    
+    public enum IntakeState{
+        EXTEND(DoubleSolenoid.Value.kForward),
+        RETRACT(DoubleSolenoid.Value.kReverse);
+
+        public DoubleSolenoid.Value value;
+
+        private IntakeState(DoubleSolenoid.Value value) {
+            this.value = value;
+        }
+    }
+    
+
+
+    // Define Hardware
+
+    private final WPI_TalonFX motor1;
+
+    private final WPI_TalonFX motor2;
+
+    private final DoubleSolenoid solenoid1;
+
+    //private final DoubleSolenoid solenoid2;
+
+    //private final ColorSensorV3 colorSensor = new colorSensorV3(i2cPort);
+
+    private IntakeMotorState intakeMotorState;
+
+    private IntakeSolenoidState intakeSolenoidState;
+
+    // CONSTRUCTOR!
+    public IntakeSubsystem(WPI_TalonFX intakeMotor1, WPI_TalonFX intakeMotor2, DoubleSolenoid intakeSolenoid1) {
+        this.motor1 = intakeMotor1;
+        this.motor2 = intakeMotor2;
+
+        this.motor2.setInverted(true);
```

Comment:
Please also configure the motors so they're free spinning, have reasonable current limits, etc.  See https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#configuration and [configAllSettings()](https://store.ctr-electronics.com/content/api/java/html/classcom_1_1ctre_1_1phoenix_1_1motorcontrol_1_1can_1_1_talon_f_x.html#ad818e523340061302d50b3014e75d8ce)